modules/base/rendering/screenspacetimevaryingimageonline.cpp

```diff
@@ -0,0 +1,234 @@
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/screenspacetimevaryingimageonline.h>
+
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/json.h>
+#include <openspace/util/timemanager.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/io/texture/texturereader.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/opengl/texture.h>
+#include <fstream>
+
+namespace {
+    constexpr std::string_view _loggerCat = "ScreenSpaceTimeVaryingImageOnline";
+
+    constexpr openspace::properties::Property::PropertyInfo JsonFileInfo = {
+        "JsonFilePath",
+        "JSON File Path",
+        "The file path to the JSON data.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    // This `ScreenSpaceRenderable` displays an image based on the current in-game
+    // simulation time. The image shown is selected from a JSON file containing
+    // timestamp-URL pairs. The image with the closest timestamp before or equal to the
+    // current time is displayed.
+    // Example JSON format:
+    // {
+    //   "files": [
+    //     {"timestamp": "2024-05-10 15:00:00.0","url": "https://example.com/image1.png"},
+    //     {"timestamp": "2024-05-10 15:02:00.0","url": "https://example.com/image2.png"}
+    //   ]
+    // }
+    struct [[codegen::Dictionary(ScreenSpaceTimeVaryingImageOnline)]] Parameters {
+        // [[codegen::verbatim(JsonFileInfo.description)]]
+        std::string jsonFilePath;
+    };
+#include "screenspacetimevaryingimageonline_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+ScreenSpaceTimeVaryingImageOnline::ScreenSpaceTimeVaryingImageOnline(
+                                                      const ghoul::Dictionary& dictionary)
+    : ScreenSpaceRenderable(dictionary)
+    , _jsonFilePath(JsonFileInfo, "")
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+
+    _jsonFilePath = absPath(p.jsonFilePath).string();
+    addProperty(_jsonFilePath);
+
+    _jsonFilePath.onChange([this]() {
+        loadJsonData(_jsonFilePath);
+        });
+}
+
+documentation::Documentation ScreenSpaceTimeVaryingImageOnline::Documentation() {
+    return codegen::doc<Parameters>("base_screenspace_time_varying_image_online");
+}
+
+bool ScreenSpaceTimeVaryingImageOnline::initialize() {
+    loadJsonData(_jsonFilePath.value());
+    return ScreenSpaceRenderable::initialize();
+}
+
+bool ScreenSpaceTimeVaryingImageOnline::deinitializeGL() {
+    _texture = nullptr;
+    return ScreenSpaceRenderable::deinitializeGL();
+}
+
+void ScreenSpaceTimeVaryingImageOnline::loadJsonData(const std::string& path) {
+    std::ifstream file(path);
```

Comment:
```suggestion
    std::ifstream file = std::ifstream(path);
```

Has the same effect, but is a bit more explicit